src/hera/events/service.py

```diff
@@ -47,6 +47,7 @@ def __init__(
         token: Optional[str] = None,
         client_certs: Optional[Tuple[str, str]] = None,
         namespace: Optional[str] = None,
+        use_session: Optional[bool] = None,
```

Comment:
Would it not be more flexible to pass in a `session` directly? (Same for WorkflowsService.)